processor/src/trace/mod.rs

```diff
@@ -56,7 +51,7 @@ pub struct ExecutionTrace {
     meta: Vec<u8>,
     layout: TraceLayout,
     main_trace: ColMatrix<Felt>,
```

Comment:
We should have `main_trace` be a `MainTrace` (and implement `AsRef<ColMatrix<Felt>>` for `MainTrace`). This is really the type that we want in the VM, and we can convert it in a `ColMatrix` for winterfell's `Trace::main_segment()`

Reply:
Makes sense, but let's do it in another PR

Reply:
Created #1201